mage_ai/api/resources/PipelineRunResource.py

```diff
@@ -114,20 +130,44 @@ async def process_collection(self, query_arg, meta, user, **kwargs):
         limit = int(meta.get(META_KEY_LIMIT, self.DEFAULT_LIMIT))
         offset = int(meta.get(META_KEY_OFFSET, 0))
 
+        include_pipeline_uuids = query_arg.get('include_pipeline_uuids', [False])
+        if include_pipeline_uuids:
+            include_pipeline_uuids = include_pipeline_uuids[0]
+
+        pipeline_tags = query_arg.get('pipeline_tag[]', [None])
+        if pipeline_tags:
+            pipeline_tags = pipeline_tags[0]
+
         pipeline_type = query_arg.get('pipeline_type', [None])
         if pipeline_type:
             pipeline_type = pipeline_type[0]
 
-        if pipeline_type is not None:
-            pipeline_type_by_pipeline_uuid = dict()
+        filter_by_pipeline_type = pipeline_type is not None
+        filter_by_pipeline_tag = pipeline_tags is not None
+        if filter_by_pipeline_type or filter_by_pipeline_tag:
             try:
+                pipeline_type_by_pipeline_uuid = dict()
+                if filter_by_pipeline_tag:
+                    pipeline_tag_cache = TagCache()
+                    pipeline_tags_by_pipeline_uuid = \
```

Comment:
This way of filtering pipeline runs by tags is not efficient.
Here is the recommend way
* Filter the pipelines by tags using the TagCache and get the pipeline uuids
* Filter the pipeline runs by the pipeline uuids

Reply:
Updated query based on what you suggested